### PR TITLE
Add `continue` in json error handler

### LINKED ISSUE #125: Event loop continues if JSON from kafka is non-parseable?
Based on https://github.com/kbase/index_runner_deluxe/blob/develop/src/index_runner/main.py#L112-L121, it appears that if the JSON received from Kafka is unparseable, the event loop will log a message but keep going as if nothing is wrong, and `_handle_msg` will be called with the `Message` class returned from the kafka client rather than a dict unmarshalled from the JSON as expected.

### PR DESCRIPTION
**Pull request by jayrbolton**
Closes #125 

I took a quick stab at adding a test for this, but then decided it wasn't worth the effort. Not the easiest thing to test, unless you have an idea about it.

### CODE REVIEW

**Comment by MrCreosote** [comment]
Actually I think if I abstract out the event loop it'll be pretty easy to unit test it, since we can just pass in the consumer and event handler and can therefore mock those out.

File: src/index_runner/main.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def main():
         except ValueError as err:
             logger.error(f'JSON parsing error: {err}')
             logger.error(f'Message content: {val}')
+            continue
```
Comment (MrCreosote) [inline src/index_runner/main.py:119]: If you just continue you'll get the same message again, won't you, since you haven't committed the offset?

Comment (MrCreosote) [reply src/index_runner/main.py:119]: aaand you beat me to it

File: src/index_runner/main.py [hunk 2]
```diff
@@ -108,13 +108,16 @@ def main():
                 logger.info('End of stream.')
             else:
                 logger.error(f"Kafka message error: {msg.error()}")
+            consumer.commit()
```
Comment (MrCreosote) [inline src/index_runner/main.py:111]: Hmm, this will also commit if the error is a network partition, which may not be what you want.

Comment (jayrbolton) [reply src/index_runner/main.py:111]: Yeah I'm seeing this does not make sense for an ERR__PARTITION_EOF. 
I may move it into the else block, but I'm still not sure that makes sense. Trying to find docs about it.

Comment (jayrbolton) [reply src/index_runner/main.py]: Reverted this for now

**Pull request merged**